src/unistd/src/getopt.rs

```diff
@@ -0,0 +1,142 @@
+//! getopt implementation for Redox, following http://pubs.opengroup.org/onlinepubs/009695399/functions/getopt.html
+
+use super::platform::types::*;
+use super::platform;
+use super::stdio;
+use super::string;
+use core::ptr;
+
+#[no_mangle]
+pub static mut optarg: *mut c_char = ptr::null_mut();
+
+#[no_mangle]
+pub static mut optind: c_int = 1;
+
+#[no_mangle]
+pub static mut opterr: c_int = 1;
+
+#[no_mangle]
+pub static mut optopt: c_int = -1;
+
+static mut current_opt: *mut c_char = ptr::null_mut();
+
+#[no_mangle]
+pub unsafe extern "C" fn getopt(
+    argc: c_int,
+    argv: *const *mut c_char,
+    optstring: *const c_char,
+) -> c_int {
+    if current_opt == ptr::null_mut() || *current_opt == 0 {
+        let current_arg = *argv.offset(optind as isize);
```

Comment:
if `argc` is 1 this could be out of bounds right?

Reply:
Good catch.  I'll fix this later today.